src/style-spec/function/convert.js

```diff
@@ -36,7 +36,9 @@ function convertIdentityFunction(parameters, propertySpec): Array<mixed> {
     const get = ['get', parameters.property];
 
     if (parameters.default === undefined) {
-        return get;
+        // By default, expressions for string-valued properties get coerced. To preserve
+        // legacy function semantics, insert an explicit assertion instead.
+        return propertySpec.type === 'string' ? ['string', get] : get;
```

Comment:
Should the `to-string` expression be dropped here: https://github.com/mapbox/mapbox-gl-js/pull/7280/files#diff-4b96c63e8fcfc96468a92c7a73e53b5eR229

Reply:
👍 , updated.